<!DOCTYPE UI><UI version="4.0">
<class>PreferencesBase</class>
<comment>****************************************************************************
**
** Copyright (C) 1992-2007 Trolltech AS. All rights reserved.
**
** This file is part of the QSA of the Qt Toolkit.
**
** For QSA Commercial License Holders (non-open source):
** 
** Licensees holding a valid Qt Script for Applications (QSA) License Agreement
** may use this file in accordance with the rights, responsibilities and
** obligations contained therein.  Please consult your licensing agreement or
** contact sales@trolltech.com if any conditions of the Licensing Agreement are
** not clear to you.
** 
** Further information about QSA licensing is available at:
** http://www.trolltech.com/products/qsa/licensing.html or by contacting
** info@trolltech.com.
** 
** 
** For Open Source Edition:  
** 
** This file may be used under the terms of the GNU General Public License
** version 2 as published by the Free Software Foundation and appearing in the
** file LICENSE.GPL included in the packaging of this file.  Please review the
** following information to ensure GNU General Public Licensing requirements
** will be met:  http://www.trolltech.com/products/qt/opensource.html 
** 
** If you are unsure which license is appropriate for your use, please review
** the following information:
** http://www.trolltech.com/products/qsa/licensing.html or contact the 
** sales department at sales@trolltech.com.

**
** This file is provided AS IS with NO WARRANTY OF ANY KIND, INCLUDING THE
** WARRANTY OF DESIGN, MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE.
**
****************************************************************************</comment>
<widget class="QWidget" name="PreferencesBase">
    <property name="geometry">
        <rect>
            <x>0</x>
            <y>0</y>
            <width>362</width>
            <height>396</height>
        </rect>
    </property>
    <property name="windowTitle">
        <string>Form1</string>
    </property>
    <layout class="QGridLayout">
        <property name="margin">
            <number>6</number>
        </property>
        <property name="spacing">
            <number>6</number>
        </property>
        <item row="0" column="0" rowspan="1" colspan="2">
        <widget class="QGroupBox" name="GroupBox2">
            <property name="title">
                <string>S&amp;yntax Highlighting</string>
            </property>
            <layout class="QGridLayout">
                <property name="margin">
                    <number>11</number>
                </property>
                <property name="spacing">
                    <number>6</number>
                </property>
                <item row="0" column="0">
                <widget class="QLabel" name="TextLabel1">
                    <property name="text">
                        <string>&amp;Element:</string>
                    </property>
                    <property name="buddy">
                        <cstring>listElements</cstring>
                    </property>
                </widget>
                </item>
                <item row="1" column="1" rowspan="3" colspan="1">
                <widget class="QWidget" name="Layout1">
                    <layout class="QGridLayout">
                        <property name="margin">
                            <number>0</number>
                        </property>
                        <property name="spacing">
                            <number>6</number>
                        </property>
                        <item row="2" column="0" rowspan="1" colspan="2">
                        <widget class="QCheckBox" name="checkBold">
                            <property name="text">
                                <string>&amp;Bold</string>
                            </property>
                        </widget>
                        </item>
                        <item row="5" column="1">
                        <widget class="QToolButton" name="buttonColor">
                            <property name="sizePolicy">
                                <sizepolicy>
                                    <hsizetype>1</hsizetype>
                                    <vsizetype>0</vsizetype>
                                    <horstretch>0</horstretch>
                                    <verstretch>0</verstretch>
                                </sizepolicy>
                            </property>
                            <property name="text">
                                <string>...</string>
                            </property>
                        </widget>
                        </item>
                        <item row="0" column="1">
                        <widget class="QComboBox" name="comboFamily">
                            <property name="editable">
                                <bool>true</bool>
                            </property>
                        </widget>
                        </item>
                        <item row="0" column="0">
                        <widget class="QLabel" name="TextLabel2">
                            <property name="text">
                                <string>&amp;Family:</string>
                            </property>
                            <property name="buddy">
                                <cstring>comboFamily</cstring>
                            </property>
                        </widget>
                        </item>
                        <item row="4" column="0" rowspan="1" colspan="2">
                        <widget class="QCheckBox" name="checkUnderline">
                            <property name="text">
                                <string>&amp;Underline</string>
                            </property>
                        </widget>
                        </item>
                        <item row="1" column="1">
                        <widget class="QSpinBox" name="spinSize">
                            <property name="minimum">
                                <number>1</number>
                            </property>
                        </widget>
                        </item>
                        <item row="3" column="0" rowspan="1" colspan="2">
                        <widget class="QCheckBox" name="checkItalic">
                            <property name="text">
                                <string>&amp;Italic</string>
                            </property>
                        </widget>
                        </item>
                        <item row="5" column="0">
                        <widget class="QLabel" name="TextLabel4">
                            <property name="text">
                                <string>Change co&amp;lor:</string>
                            </property>
                            <property name="buddy">
                                <cstring>buttonColor</cstring>
                            </property>                            
                        </widget>
                        </item>
                        <item row="1" column="0">
                        <widget class="QLabel" name="TextLabel3">
                            <property name="text">
                                <string>&amp;Size:</string>
                            </property>
                            <property name="buddy">
                                <cstring>spinSize</cstring>
                            </property>
                        </widget>
                        </item>
                    </layout>
                </widget>
                </item>
                <item row="1" column="0">
                <widget class="QListWidget" name="listElements">
                    <item>
                        <property name="text">
                            <string>Comment</string>
                        </property>
                    </item>
                    <item>
                        <property name="text">
                            <string>Number</string>
                        </property>
                    </item>
                    <item>
                        <property name="text">
                            <string>String</string>
                        </property>
                    </item>
                    <item>
                        <property name="text">
                            <string>Type</string>
                        </property>
                    </item>
                    <item>
                        <property name="text">
                            <string>Keyword</string>
                        </property>
                    </item>
                    <item>
                        <property name="text">
                            <string>Preprocessor</string>
                        </property>
                    </item>
                    <item>
                        <property name="text">
                            <string>Label</string>
                        </property>
                    </item>
                    <item>
                        <property name="text">
                            <string>Standard</string>
                        </property>
                    </item>
                </widget>
                </item>
                <item row="2" column="0">
                <widget class="QLabel" name="TextLabel2_2">
                    <property name="text">
                        <string>&amp;Preview:</string>
                    </property>
                    <property name="buddy">
                        <cstring>editPreview</cstring>
                    </property>
                </widget>
                </item>
                <item row="3" column="0">
                <widget class="QLineEdit" name="editPreview">
                    <property name="text">
                        <string>Some Text</string>
                    </property>
                </widget>
                </item>
                <item row="4" column="1">
                <spacer name="Spacer2">
                    <property name="orientation">
                        <enum>Vertical</enum>
                    </property>
                    <property name="sizeType">
                        <enum>Expanding</enum>
                    </property>
                    <property name="sizeHint">
                        <size>
                            <width>20</width>
                            <height>0</height>
                        </size>
                    </property>
                </spacer>
                </item>
            </layout>
        </widget>
        </item>
        <item row="1" column="0">
        <widget class="QGroupBox" name="GroupBox3">
            <property name="title">
                <string>Optio&amp;ns</string>
            </property>
            <layout class="QVBoxLayout">
                <property name="margin">
                    <number>11</number>
                </property>
                <property name="spacing">
                    <number>6</number>
                </property>
                <item>
                <widget class="QCheckBox" name="checkWordWrap">
                    <property name="text">
                        <string>&amp;Word Wrap</string>
                    </property>
                </widget>
                </item>
                <item>
                <widget class="QCheckBox" name="checkCompletion">
                    <property name="text">
                        <string>Comple&amp;tion</string>
                    </property>
                </widget>
                </item>
                <item>
                <widget class="QCheckBox" name="checkParenMatching">
                    <property name="text">
                        <string>Parentheses &amp;Matching</string>
                    </property>
                </widget>
                </item>
            </layout>
        </widget>
        </item>
        <item row="1" column="1">
        <widget class="QGroupBox" name="GroupBox3_2">
            <property name="title">
                <string>Indentation</string>
            </property>
            <layout class="QVBoxLayout">
                <property name="margin">
                    <number>11</number>
                </property>
                <property name="spacing">
                    <number>6</number>
                </property>
                <item>
                <widget class="QWidget" name="Layout2">
                    <layout class="QGridLayout">
                        <property name="margin">
                            <number>0</number>
                        </property>
                        <property name="spacing">
                            <number>6</number>
                        </property>
                        <item row="0" column="1">
                        <widget class="QSpinBox" name="spinTabSize">
                            <property name="value">
                                <number>8</number>
                            </property>
                        </widget>
                        </item>
                        <item row="0" column="0">
                        <widget class="QLabel" name="TextLabel1_2">
                            <property name="text">
                                <string>Tab Size:</string>
                            </property>
                        </widget>
                        </item>
                        <item row="1" column="1">
                        <widget class="QSpinBox" name="spinIndentSize">
                            <property name="value">
                                <number>4</number>
                            </property>
                        </widget>
                        </item>
                        <item row="1" column="0">
                        <widget class="QLabel" name="TextLabel2_3">
                            <property name="text">
                                <string>Indent Size:</string>
                            </property>
                        </widget>
                        </item>
                    </layout>
                </widget>
                </item>
                <item>
                <widget class="QCheckBox" name="checkKeepTabs">
                    <property name="text">
                        <string>Keep Tabs</string>
                    </property>
                    <property name="checked">
                        <bool>true</bool>
                    </property>
                </widget>
                </item>
                <item>
                <widget class="QCheckBox" name="checkAutoIndent">
                    <property name="text">
                        <string>Auto Indent</string>
                    </property>
                    <property name="checked">
                        <bool>true</bool>
                    </property>
                </widget>
                </item>
            </layout>
        </widget>
        </item>
    </layout>
</widget>
<connections>
    <connection>
        <sender>spinSize</sender>
        <signal>valueChanged(int)</signal>
        <receiver>PreferencesBase</receiver>
        <slot>sizeChanged(int)</slot>
    </connection>
    <connection>
        <sender>checkBold</sender>
        <signal>toggled(bool)</signal>
        <receiver>PreferencesBase</receiver>
        <slot>boldChanged(bool)</slot>
    </connection>
    <connection>
        <sender>checkItalic</sender>
        <signal>toggled(bool)</signal>
        <receiver>PreferencesBase</receiver>
        <slot>italicChanged(bool)</slot>
    </connection>
    <connection>
        <sender>checkUnderline</sender>
        <signal>toggled(bool)</signal>
        <receiver>PreferencesBase</receiver>
        <slot>underlineChanged(bool)</slot>
    </connection>
    <connection>
        <sender>buttonColor</sender>
        <signal>clicked()</signal>
        <receiver>PreferencesBase</receiver>
        <slot>colorClicked()</slot>
    </connection>
    <connection>
        <sender>comboFamily</sender>
        <signal>activated(const QString&amp;)</signal>
        <receiver>PreferencesBase</receiver>
        <slot>familyChanged(const QString&amp;)</slot>
    </connection>
    <connection>
        <sender>listElements</sender>
        <signal>currentTextChanged(const QString &amp;)</signal>
        <receiver>PreferencesBase</receiver>
        <slot>elementChanged(const QString &amp;)</slot>
    </connection>
</connections>
<tabstops>
    <tabstop>listElements</tabstop>
    <tabstop>editPreview</tabstop>
    <tabstop>comboFamily</tabstop>
    <tabstop>spinSize</tabstop>
    <tabstop>checkBold</tabstop>
    <tabstop>checkItalic</tabstop>
    <tabstop>checkUnderline</tabstop>
    <tabstop>checkWordWrap</tabstop>
    <tabstop>checkCompletion</tabstop>
    <tabstop>checkParenMatching</tabstop>
</tabstops>
<variables>
    <variable>QMap&lt;QString, ConfigStyle&gt; styles;</variable>
    <variable>ConfigStyle currentStyle;</variable>
    <variable>QString currentElement;</variable>
</variables>
<slots>
    <slot>init()</slot>
    <slot>destroy()</slot>
    <slot>colorClicked()</slot>
    <slot>reInit()</slot>
    <slot>save()</slot>
    <slot>updatePreview()</slot>
    <slot>boldChanged( bool b )</slot>
    <slot>elementChanged( const QString &amp; element )</slot>
    <slot>familyChanged( const QString &amp; f )</slot>
    <slot>italicChanged( bool b )</slot>
    <slot>setColorPixmap( const QColor &amp; c )</slot>
    <slot>sizeChanged( int s )</slot>
    <slot>underlineChanged( bool b )</slot>
</slots>
<layoutdefaults spacing="6" margin="11"/>
</UI>
